README.md

```diff
@@ -117,19 +103,90 @@ The default newsletter template is:
 
+---
+
+## Routers
+<!-- Note. This will be subject to change once the fs readdir router system is implemented. -->
+
+The Express app's router is set up by `/src/route.js`. The file route.js imports routers from the `/routes` folder. Each router file has the following format:
+
+```js
+const express = require('express');
+const router = express.Router();
+// This router is used to configure the app for a specific route
+
+// GET requests
+app.get('/path', (req, res, next) => { 
+	// some code
+	return res.renderFile(template, ctx);
+});
+
+// POST requests
+app.post('/path', (req, res, next) => { 
+	// some code
+	return res.status(statusCode).send;
+});
+
+/* 
+Notes:
+	The next argument is optional in most cases. 
+	The functions in the requests can be asynchronous too.
+	You can use other routers as well if needed.
+*/
+
+module.exports = router;
+```
+
+---
+
+## Contributing
+
+The owner of this repository is @PartMan7, who also runs the server. He's also running the server cost (at the time of writing he's in his final year, but it'll carry on even after graduation).
+
+All non-trivial changes are done through PULL REQUESTS ONLY. The WebDev Team Head (currently @Goose-Of-War) is responsible for testing and merging all PRs. Feel free to pester them to look at the changes you've prepared.
+
+To create a pull request, navigate to the `dev` branch (clicking on the GitHub client, or `git checkout dev` on the CLI) and create a new branch based on it (`New Branch` button (based on the `dev` branch, again!) on the GitHub client's branches page, or `git checkout -b [branch-name]` on the CLI). Pull requests will _never_ be merged directly to `main`; they will be first merged to `dev` and batches of changes and/or patches will be merged from `dev` to `main` alongside version increments.
+
+To work on templates which require access to credentials using `dev userless` (`du`) mode, check out [this page](/docs/TESTING.md) for some info.
+
+Changes to the `dev` server are automatically deployed to [test.kgpmask.club](https://test.kgpmask.club). This does not extend to environment configuration changes (eg: docker changes).
+
+Ensure that pull requests pass tests (`npm test` for both lint and mocha tests).
+
+---
+
+## Guidelines
+
+- For any new route, make sure to use a new router file instead of adding to `route.js`
+- Make sure to extend your template from either `_base.njk` or another template (in case of similar pages, like events or newsletters).
+- Make sure the code passes the lint and mocha tests. You can check that by running `npm run test`. The ESLint configuration can be seen in `.eslintrc.json` and tests can be seen in the `test` folder.
+- If you have any questions, post 'em in the WebDev channel.
+- If you aren't a member of the society but have queries/reports/suggestions, feel free to use the [Issues](https://github.com/kgpmask/MASK/issues)/[Discussions](https://github.com/kgpmask/MASK/discussions) pages.
+
+---
+
 ## Credits:
```

Comment:
i can't find the official source, but there should be no trailing punctuation after headings, or so as my linter says
keep them as is ig

Reply:
But where heading?

Reply:
Guidelines had punctuations?

Reply:
"Credits"